src/lib/comms/sol-socket.c

```diff
@@ -51,6 +51,7 @@ sol_socket_ip_new(const struct sol_socket_options *options)
 #ifdef DTLS
         s = sol_socket_default_dtls_new(options);
 #else
+        errno = ENOSYS;
         SOL_WRN("DTLS is not enabled, secure socket is not possible");
         return NULL;
```

Comment:
ok, but you should also check `#ifdef DTLS` before trying to set the secure flag in coap/oic.